FireRender.Maya.Src/FireRenderHairs.cpp

```diff
@@ -547,6 +548,80 @@ void FireRenderHair::detachFromScene()
 	m_isVisible = false;
 }
 
+void FireRenderHair::setPrimaryVisibility(bool primaryVisibility)
+{
+	for (frw::Curve& curve : m_Curves)
+	{
+		curve.SetPrimaryVisibility(primaryVisibility);
+	}
+}
+
+void FireRenderHair::setReflectionVisibility(bool reflectionVisibility)
+{
+	for (frw::Curve& curve : m_Curves)
+	{
+		curve.SetReflectionVisibility(reflectionVisibility);
+	}
+}
+
+void FireRenderHair::setRefractionVisibility(bool refractionVisibility)
+{
+	for (frw::Curve& curve : m_Curves)
+	{
+		curve.setRefractionVisibility(refractionVisibility);
+	}
+}
+
+void FireRenderHair::setCastShadows(bool castShadow)
+{
+	for (frw::Curve& curve : m_Curves)
+	{
+		curve.SetShadowFlag(castShadow);
+	}
+}
+
+void FireRenderHair::setRenderStats(MDagPath dagPath)
+{
+	if (!dagPath.isValid())
+		return;
+
+	MFnDependencyNode depNode(dagPath.node());
+
+	MPlug visibleInReflectionsPlug = depNode.findPlug("visibleInReflections");
+	if (!visibleInReflectionsPlug.isNull())
+	{
+		MString dbgName = visibleInReflectionsPlug.name();
+
+		bool visibleInReflections;
```

Comment:
pls initialize this bool with something. I'm afraid what happens if getValue does nothing for some reason